oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/mx/FieldsHelper.java

```diff
@@ -82,7 +85,24 @@ public void init(final InputStream inputStream,
             final StringBuffer serviceNamePattern = new StringBuffer();
             while (m.find()) {
                 final String property = m.group("property");
-                flatBuffersFieldNames.add(Splitter.on('.').omitEmptyStrings().splitToList(property));
+                List<String> tokens = Splitter.on('.').omitEmptyStrings().splitToList(property);
```

Comment:
Just did a quick Google search, replacing the `.` with this delimiter `\.(?=([^"]*"[^"]*")*[^"]*$)` can do the trick, but you can make the call whether to replace it or not

Reply:
I prefer the current way. It is more clear.